src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -668,6 +674,43 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
                         "Could not install root log handler on node: " + c.getName(), e);
             }
         }
+
+        @Override
+        public void onOffline(@Nonnull final Computer c, @CheckForNull OfflineCause cause) {
+            if (Boolean.getBoolean(SupportPlugin.class.getName() + ".generateOnOfflineNode")
+                    && cause != null && !cause.getClass().getName().contains("RestartCause")) {
```

Comment:
What about if `cause == null`? I would think you would want to generate a bundle in that case, right? But here you are not.